Base/QTCore/qSlicerExtensionsManagerModel.cxx

```diff
@@ -409,7 +410,15 @@ void qSlicerExtensionsManagerModelPrivate::queryExtensionsMetadataFromServer()
     parameters["app_revision"] = q->slicerRevision();
     parameters["os"] = q->slicerOs();
     parameters["arch"] = q->slicerArch();
-    parameters["offset"] = QString::number(this->ExtensionsMetadataFromServerQueryResults.size());
+    if (this->ServerQueryWithLimit)
+      {
+      parameters["offset"] = QString::number(this->ExtensionsMetadataFromServerQueryResults.size());
+      parameters["limit"] = QString::number(50); // corresponds to the default limit value also used on the server
```

Comment:
It is better to rely on the server's default, as only the server knows what limit it prefers. The client has no way of knowing what is an acceptable limit and it should not matter for the client either. So, I would remove this line, maybe just add a comment there explaining why we don't set the limit.